tempo_embeddings/embeddings/vector_database.py

```diff
@@ -0,0 +1,336 @@
+            self.embedding_function = embedding_functions.HuggingFaceEmbeddingFunction(
+                api_key=self.embedder_config.get("api_key"),
+                model_name=embedder_name
+            )
+        elif self.embedder_config and self.embedder_config.get("type") == "custom":
+            self.requires_explicit_embeddings = True
+            self.embedding_function = None
+        else:
+            self.embedding_function = None
+
+        self.config = {
+            "embedder_name": self.embedder_name,
+            "embedder_type": self.embedder_config["type"],
+            "requires_explicit_embeddings": self.requires_explicit_embeddings,
+            "existing_collections": []
+        }
+    
+    def _save_config(self):
+        if self.config:
+            json.dump(self.config, open(f"{self.db_path}/config.json", "w"), indent=2)
+
+    def connect(self):
+        if not self.client:
+            self.client = chromadb.PersistentClient(path=self.db_path)
+            # If the given path existed then we will load the records AND override the parameters with the config
+            config_path = f"{self.db_path}/config.json"
+            if os.path.exists(config_path):
+                self.config = json.load(open(config_path))
+                print(f"Database Path '{self.db_path}' already exists. Loading current configuration\n{self.config}")
+            else:
+                print(f"Creating NEW Database in {self.db_path}...")
+                self._save_config()
+        else:
+            print(f"A connection to the client already exsists in this session: {self.client}")
+
+    def create_new_collection(self, name: str, passages: List[Passage] = None, embeddings: ArrayLike = None) -> Optional[Collection]:
+        if not self.client:
+            print("Please connect to a valid database first")
+            return None
+        # Create NEW collection and Embeds the given passages. Do nothing otherwise
+        try:
+            if self.embedding_function:
+                collection = self.client.create_collection(name, 
+                                                            embedding_function=self.embedding_function,
+                                                            metadata={"hnsw:space": "cosine"})
+            else:
+                collection = self.client.create_collection(name, metadata={"hnsw:space": "cosine"})
+            self.config["existing_collections"].append(name)
+            self.active_collection_name = name
+            self._save_config()
+            # If the collection is new then insert the corresponding passages already
+            if passages: self.insert_passages_embeddings(collection, passages, embeddings)
+
+            return collection
+        except UniqueConstraintError:
+            print(f"Collection '{name}' already exists. Not doing anything. (You can delete it or use the 'get_existing_collection' method)")
+            return None
```

Comment:
Why do we need to catch this exception at all? Is there a use case in which we would like to run this with an existing collection name and silently continue?

Reply:
yes, currently there is... but perhaps is nicer to discuss the whole workflow in the notebook example so you tell me your opinions about how we can shape the code. The fact that it is a notebook confuses me about how defensive the code should be